### ISSUE TITLE
MGLFeature not detected by VoiceOver, occasionally crashing

### ISSUE THREAD

**Issue by allanweir** [mapbox/mapbox-gl-native#13922]
Hey, I've just recently switched a project I work on over to Mapbox from Apple Maps. At the same time we're adding system-wide accessibility improvements to the project. As part of the Mapbox integration I dynamically create MGLSymbolStyleLayer from MGLShapeSource and MGLShape. This is from GeoJSON created from API data. The data shows perfectly and works with a UITapGestureRecognizer to find features at a point when tapped. However when I turn on Voiceover, it is either unresponsive or crashes. My issue might be a bad implementation but it's hard to tell as the documentation makes no obvious reference to accessibility. If this turns out to be a bug or two I can report separately.

### Steps to reproduce

 1. Create a single MGLShape from a GeoJSON Feature Collection. Adding name_en as an attribute to each point - I've picked this up from the source code in MGLMapAccessibilityElement.mm
 2. Create a MGLShapeSource with this data (clustered, if that's relevant?)
 3. Create two MGLSymbolStyleLayer's, one for clustering and one for not
 4. Either start app with VoiceOver enabled - open map and try to select feature, the app leaks memory until it crashes. Instruments shows 370MB of CFNumber references, created by NSPlaceholderNumber on an Accessibility method.
5. Start app with VoiceOver not enabled, enable from quick menu on the phone, open the map and some roads are selectable on the map but it's not possible to access anything else. UI will start to slow down and crash at a slower pace - sometimes the same as this crash https://github.com/mapbox/mapbox-gl-native/issues/10994

### Expected behaviour

VoiceOver should be able to find the objects from the added layer, with their names read out and features selectable. Possibly documentation directs you on the best way to do this.

### Actual behaviour

Features aren't detected by VoiceOver. App crashes from memory leak or https://github.com/mapbox/mapbox-gl-native/issues/10994

![screenshot 2019-02-13 at 16 08 19](https://user-images.githubusercontent.com/2892400/52731236-23ef9280-2fb5-11e9-9f7d-977aa7c3f22c.png)

### Configuration

**Mapbox SDK versions: 4.8** 
**iOS/macOS versions: 12.1** 
**Device/simulator models: iPhone X (Device) / iPhone 8 (Simulator)**
**Xcode version: 10.1**

**Comment by julianrex** [comment]
Hi @allanweir - we're aware of some poor performance when accessibility is enabled, in addition to the crash you reference - so many thanks for your detailed report that will help us track down these issues.

**Comment by jdhuggins** [comment]
Is there a work around or fix for this?

**Comment by MikeFox87** [comment]
We've noticed our own apps are crashing within a minute if we enable VoiceOver and start tapping on the map. We're currently on Mapbox 4.8.0.
